### PR TITLE
Multibuy 1.5

### PR DESCRIPTION
**Pull request by IAmMorrow**
## 🦒 Context (issues, jira)

https://ledgerhq.atlassian.net/browse/LIVE-717

## 💻  Description / Demo (image or video)

This PR enable the MultiBuy 1.5 implementation including:
- New platform app providers
- Dynamic links to buy in accounts, market and asset pages
- Replaced buy section with Ramp catalog "multibuy" 

## 🖤  Expectations to reach

PR must pass CI, rebase develop if conflicts. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [ ] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [ ] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [x] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

### CODE REVIEW

File: src/renderer/components/PerCurrencySelectAccount/state.js [hunk 1]
```diff
@@ -55,24 +55,29 @@ export type UseCurrencyAccountSelectReturnType = {
...
+      : allCurrencies.length > 0
+      ? allCurrencies[0]
+      : undefined;
     if (!currency) {
       return { currency: null, accountId: null };
     }
     const availableAccounts = getAccountTuplesForCurrency(currency, allAccounts, hideEmpty);
-    const { accountId } = defaultAccount
-      ? { accountId: defaultAccount.id }
+    console.log({ availableAccounts });
```
Comment (LFBarreto) [inline src/renderer/components/PerCurrencySelectAccount/state.js:78]: :hankey:

File: src/renderer/screens/account/AccountHeaderActions.js [hunk 2]
```diff
@@ -103,8 +104,28 @@ const AccountHeaderActions = ({ account, parentAccount, openModal, t }: Props) =
...
+    const allSellableCryptoCurrencyIds = getAllSupportedCryptoCurrencyIds(
+      rampCatalog.value.offRamp,
+    );
+    return [
+      allBuyableCryptoCurrencyIds.includes(currency.id),
+      allSellableCryptoCurrencyIds.includes(currency.id),
+    ];
+  }, [rampCatalog.value, currency.id]);
+
+  console.log({
```
Comment (LFBarreto) [inline src/renderer/screens/account/AccountHeaderActions.js:124]: :hankey:

File: src/renderer/screens/exchange/Buy/index.js [hunk 3]
```diff
@@ -1,114 +1,113 @@
...
-const ContinueButton: ThemedComponent<{}> = styled(Button).attrs(() => ({}))`
-  padding: 12px 16px;
-`;
+    currenciesByMarketcap(filteredCurrencies).then(sortedCurrencies => {
+      setSortedCurrencies(sortedCurrencies);
+    });
+  }, []);
 
-type Provider = "moonpay" | "coinify" | null;
+  console.log({ sortedCurrencies });
```
Comment (LFBarreto) [inline src/renderer/screens/exchange/Buy/index.js:42]: :hankey:

File: src/renderer/screens/market/MarketCoinScreen/index.tsx [hunk 4]
```diff
@@ -84,8 +86,27 @@ export default function MarketCoinScreen() {
...
+    }
+
+    const onRampProviders = filterRampCatalogEntries(rampCatalog.value.onRamp, {
+      tickers: [currency.ticker],
+    });
+    const offRampProviders = filterRampCatalogEntries(rampCatalog.value.offRamp, {
+      tickers: [currency.ticker],
+    });
+
+    console.log(onRampProviders, currency);
```
Comment (LFBarreto) [inline src/renderer/screens/market/MarketCoinScreen/index.tsx:102]: :hankey:

File: src/renderer/screens/platform/App.js [hunk 5]
```diff
@@ -23,28 +25,30 @@ export default function PlatformApp({ match }: Props) {
...
   const { state: urlParams } = useLocation();
   const { appId } = match.params;
-  const { manifests } = usePlatformApp();
-  const manifest = manifests.get(appId);
+  const localManifest = useLocalLiveAppManifest(appId);
+  const remoteManifest = useRemoteLiveAppManifest(appId);
+
+  const manifest = localManifest || remoteManifest;
+
+  console.log(appId, manifest);
```
Comment (LFBarreto) [inline src/renderer/screens/platform/App.js:33]: :hankey:

**Review by LFBarreto — CHANGES_REQUESTED** [review]
Meh :laughing: 
remove console log and do a quick lint pls :pray:

**Review by gre — CHANGES_REQUESTED** [review]
lint issue

**Pull request merged**